.github/workflows/citgm-package.yml

```diff
@@ -0,0 +1,229 @@
+name: CITGM Package
+
+on:
+  workflow_dispatch:
+    inputs:
+      package:
+        description: 'Package to test'
+        required: true
+        type: string
+      node-version:
+        description: 'Node version to test'
+        required: true
+        type: string
+        default: '20'
+      os:
+        description: 'Operating System'
+        required: false
+        type: choice
+        default: 'ubuntu-latest'
+        options:
+          - 'ubuntu-latest'
+          - 'windows-latest'
+          - 'macos-latest'
+      mongodb:
+        description: 'Mongodb'
+        required: false
+        type: boolean
+        default: false
+      redis:
+        description: 'Redis'
+        required: false
+        type: boolean
+        default: false
+      kafka:
+        description: 'Kafka'
+        required: false
+        type: boolean
+        default: false
+      elasticsearch:
+        description: 'Elasticsearch'
+        required: false
+        type: boolean
+        default: false
+      mysql:
+        description: 'Mysql'
+        required: false
+        type: boolean
+        default: false
+      postgres:
+        description: 'Postgres'
+        required: false
+        type: boolean
+        default: false
+  workflow_call:
+    inputs:
+      package:
+        description: 'Package to test'
+        required: true
+        type: string
+      node-version:
+        description: 'Node version to test'
+        required: true
+        type: string
+        default: '20'
+      os:
+        description: 'Operating System'
+        required: false
+        type: string
+        default: 'ubuntu-latest'
+      mongodb:
+        description: 'Mongodb'
+        required: false
+        type: boolean
+        default: false
+      redis:
+        description: 'Redis'
+        required: false
+        type: boolean
+        default: false
+      kafka:
+        description: 'Kafka'
+        required: false
+        type: boolean
+        default: false
+      elasticsearch:
+        description: 'Elasticsearch'
+        required: false
+        type: boolean
+        default: false
+      mysql:
+        description: 'Mysql'
+        required: false
+        type: boolean
+        default: false
+      postgres:
+        description: 'Postgres'
+        required: false
+        type: boolean
+        default: false
+      
+jobs:
+  core-plugins:
+    name: CITGM
+    runs-on: ${{inputs.os}}
+    permissions:
+      contents: read
+    services:
```

Comment:
I feel maintaining this at fastify repo level and plugin level can be a pain. Can't we standardize to use a `docker-compose.yml` file in repositories which need it? Like having a command `stat docker-compose.yml && docker-compose up -d --wait` (for example)

Reply:
+1 on @zekth but I'm also over the fence about having `docker-compose.yml` as we move the maintenance to each library to match the `CITGM` flow of fastify.

If the plan is to just support `core plugins` for now, shall we consider maybe having a custom action in another repo to offload that maintenance somewhere?

In that way, we can also (maybe) empower the community to add their plugins to be tested (e.g. `platformatic`, `opentelemetry/fastify` just to name a few)